### PR TITLE
feat(datasource/sites): support sites list

### PR DESCRIPTION
**Pull request by Integralist**
Fixes part of @sjparkinson issue reported here https://github.com/signalsciences/terraform-provider-sigsci/issues/111

```terraform
terraform {
  required_providers {
    sigsci = {
      source  = "signalsciences/sigsci"
      version = "1.2.20"
    }
  }
}

resource "sigsci_site" "example1" {
  short_name   = "terraform_site1"
  display_name = "terraform test example site1"
}

resource "sigsci_site" "example2" {
  short_name   = "terraform_site2"
  display_name = "terraform test example site2"
}

resource "sigsci_site" "example3" {
  short_name   = "terraform_site3"
  display_name = "terraform test example site3"
}

data "sigsci_sites" "example" {
  depends_on = [sigsci_site.example1, sigsci_site.example2, sigsci_site.example3]
  filter     = "terraform test example site1" # checks against `name` and `display_name`
}
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,11 +1,15 @@
+clean:
+	rm -rf ./bin
 
-build:
-	go build -o terraform-provider-sigsci
+build: clean
+	go build -o bin/terraform-provider-sigsci
+	@sh -c "'$(CURDIR)/scripts/generate-dev-overrides.sh'"
```
Comment (Integralist) [inline Makefile:6]: This enables me to locally test the provider.

File: provider/datasource_sites.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+						"analytics_events_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's analytics events",
+						},
+						"blacklist_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's blacklist",
+						},
```
Comment (Integralist) [inline provider/datasource_sites.go:55]: **NOTE:** The API has a `blocklist` field but it's not implemented in the `go-sigsci` API client.

File: provider/datasource_sites.go [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+	corp := pm.Corp
+
+	// API documentation:
+	// https://docs.fastly.com/signalsciences/api/#_corps__corpName__sites_get
+	sites, err := sc.ListSites(corp)
+	if err != nil {
+		return err
+	}
+
+	d.SetId("list_sites")
```
Comment (Integralist) [inline provider/datasource_sites.go:155]: Previously I was dynamically generating a dynamic ID but opted for a static ID as I don't imagine there will be multiple instances of the `sigsci_sites` data source. If we _do_ think that is a possibility then we should make this dynamically generated (possibly using something like https://github.com/google/uuid to generate the ID).

File: provider/datasource_sites.go [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+				delete(data, k)
+			}
+		}
+		results = append(results, data)
+	}
+
+	if filter != "" {
+		for idx, site := range results {
+			if site["name"] == filter || site["display_name"] == filter {
+				return results[idx : idx+1]
```
Comment (Integralist) [inline provider/datasource_sites.go:207]: The implementation for the `filter` attribute is 'match only one', e.g. the first site that matches is returned.

If we want the filter to return multiple potential sites, then the implementation should be updated.

File: provider/lib.go [hunk 5]
```diff
@@ -38,7 +38,7 @@ func expandStringArray(entries *schema.Set) []string {
 }
 
 func flattenDetections(detections []sigsci.Detection) []interface{} {
-	var detectionsSet = make([]interface{}, len(detections))
+	detectionsSet := make([]interface{}, len(detections))
```
Comment (Integralist) [inline provider/lib.go:41]: My editor made a bunch of automated refactors.

**Review by Integralist — COMMENTED** [review]
Comments for the reviewer...

**Review by shawnps — APPROVED** [review]

**Comment by shawnps** [comment]
LGTM, thank you @Integralist!

**Pull request merged**